Assets/Scripts/Services/DiscordService.cs

```diff
@@ -0,0 +1,199 @@
+        {
+            base.OnDisconnectedFromPhoton();
+            isJoinedLobby = false;
+        }
+
+        public void UpdateDiscordActivity(global::Room room)
+        {
+            if (room.GetName().Equals("Singleplayer"))
+                UpdateSinglePlayerActivity(room);
+            else
+                UpdateMultiPlayerActivity(room);
+        }
+
+        private void InMenu()
+        {
+            var activityStruct = new Activity
+            {
+                Assets = assetsStruct,
+                State = "In Menu"
+            };
+            activityManager.UpdateActivity(activityStruct, (result) =>
+            {
+                Debug.Log(result == Result.Ok
+                    ? $"Updated to Main Menu"
+                    : $"Failed to Update to Main Menu");
+            });
+        }
+
+        private void UpdateSinglePlayerActivity(global::Room room)
+        {
+            var activityStruct = new Activity
+            {
+                Assets = assetsStruct,
+                State = "SinglePlayer",
+                Details = room.GetLevel().Name + " - " + room.GetGamemode(),
+            };
+            activityManager.UpdateActivity(activityStruct,
+                (result) =>
+                {
+                    Debug.Log(result == Result.Ok
+                        ? $"Updated Single player presence."
+                        : $"Failed to Update Single player presence.");
+                });
+        }
+
+        private void UpdateMultiPlayerActivity(global::Room room)
+        {
+            var activityStruct = new Activity
+            {
+                Assets = assetsStruct,
+                State = room.GetName() + "- [" + PhotonNetwork.CloudRegion.ToString().ToUpper() + "]",
+                Details = room.GetLevel().Name + " - " + room.GetGamemode(),
+                Party = new ActivityParty
+                {
+                    Size = new PartySize
+                    {
+                        CurrentSize = room.PlayerCount,
+                        MaxSize = room.MaxPlayers
+                    },
+                    Id = room.GetHashCode().ToString(),
+                },
+                Secrets = new ActivitySecrets
+                {
+                    Join = room.Name,
+                }
+            };
+
+            activityManager.UpdateActivity(activityStruct,
+                (result) =>
+                {
+                    Debug.Log(result == Result.Ok
+                        ? $"Updated Multi-player party."
+                        : $"Failed to Update Multi-player party.");
+                });
+        }
+
+        #region Helper Methods
+
+        private static string GetApplicationPath()
+        {
+            string path = System.Environment.CurrentDirectory;
+            path += "\\";
+            switch (Application.platform)
+            {
+                case RuntimePlatform.WindowsPlayer:
+                    path += "AoTTG 2.exe";
```

Comment:
Perhaps instead of hard coding "AoTTG 2" here, we should use `Application.companyName`. Whenever Unity does a build, it will name the .exe / .app after this value. (https://docs.unity3d.com/ScriptReference/Application-companyName.html)